src/app/services/scroll.service.ts

```diff
@@ -0,0 +1,164 @@
+let scrollServiceIsInstantiated: boolean;
+
+type BtnScrollStateType = 'click' | 'mouseover' | '';
+
+interface ScrollSettingValues {
+	name: string;
+	btnState: BtnScrollStateType;
+	bigScrollActivated: boolean;
+}
+
+class ScrollService {
+	btnScrollUp: HTMLElement = null;
+	btnScrollDown: HTMLElement = null;
+	btnState: BtnScrollStateType = '';
+	bigScrollActivated = false;
+	scrollSteps = 10;
+	scrollTimer = 50;
+
+	settingsValues: ScrollSettingValues[] = [];
+
+	constructor() {
+		if (scrollServiceIsInstantiated) {
+			throw new Error('Le ScrollService est déjà instancié.');
+		}
+
+		scrollServiceIsInstantiated = true;
+
+		this.setScrollClass();
+	}
+
+	/* Adds the style required for scrolling functions */
+	setScrollClass = (): void => {
+		let styleScroll = `
+			.cplus-big-scroll::-webkit-scrollbar, .cplus-big-scroll *::-webkit-scrollbar {
+					width: 2rem;
+			}
+			.cplus-big-scroll::-webkit-scrollbar-thumb, .cplus-big-scroll *::-webkit-scrollbar-thumb {
+				background-color: lightgrey;
+				border-radius: 1.75rem
+				width: 2rem;
+				cursor: pointer;
+			}
+			.cplus-big-scroll::-webkit-scrollbar-thumb:hover, .cplus-big-scroll *::-webkit-scrollbar-thumb:hover {
+				background-color: grey;
+			}
+
+			#cplus-container-scroll-buttons {
+				display: flex;
+				gap: 1rem;
+				position: fixed;
+				bottom: 1rem;
+				right: 1rem;
+				z-index: 2147483647;
+			}
+
+			#cplus-container-scroll-buttons button {
+				background: #f16e00;
+				color: #000;
+				border: none;
+				font-weight: bold;
+				padding: 1rem 2rem;
+			}
+			.d-none {
+				display: none;
+			}
+		`;
+
+		stylesServiceInstance.setStyle('scroll', styleScroll);
+	}
+
+	/* Allows a setting to enable or disable scroll functionality by listing the values for each setting */
+	setScroll = (values: ScrollSettingValues): void => {
+		let addNewSetting = false;
+		if (this.settingsValues.length > 0) {
+			for (let setting of this.settingsValues) {
+				if (setting.name === values.name) {
+					setting.btnState = values.btnState;
+					setting.bigScrollActivated = values.bigScrollActivated;
+					addNewSetting = false;
+					break;
```

Comment:
Le `break` est là pourquoi ? Sortir de la boucle `for` ? Je trouve ça louche :D

Reply:
Alors je crois que c'est pour ne pas repasser la variable `addNewSetting` à `true` si le réglage est déjà listé.

Reply:
Je crois que je comprends : tu boucles dans les `settingsValues` jusqu’à trouver celui qui correspond à l’argument `values` ? Et après tu interromps la boucle ?

Pour moi on a un problème de logique ici, mais je ne suis pas sûr de bien comprendre ce que tu cherches à faire.

Si je reformule : 
- si tu trouves un réglage qui correspond à l’argument `values`, tu redéfinis tes trois variables,
- et s’il n’y en a pas, tu passes `addNewSetting` à `true`.

C’est ça ?

Si oui, ne serait-il pas plus simple de cherche une clé dans `this.settingsValue` qui corresponde à `values.name` directement ? Ça t’évite la boucle.